pytentiostat/config_reader.py

```diff
@@ -81,29 +158,108 @@ def get_cv_params(config_data):
 
 
 def get_exp_type(config_data):
+    """
+    Returns the type of experiment from the config data.
+
+    Parameters
+    ----------
+    config_data : dict
+        The configuration data
+
+    Returns
+    -------
+    exp_type : str
+        The type of experiment to be run.
+    """
+    
     exp_type = config_data["general_parameters"]["experiment_type"]
 
     return exp_type
 
 
 def get_exp_time(config_data):
+    """
+    Returns how long the experiment is to be run for the chronoamperometry case.
+
+    Parameters
+    ----------
+    config_data : dict
+        The configuration data
+
+    Returns
+    -------
+    exp_time : float
+        How long the experiment will be run in seconds.
+    """
+    
     exp_time = config_data["chronoamperometry"]["time"]
 
     return exp_time
 
 
 def get_rest(config_data):
+    """
+    Returns how long to hold before experiment starts from the config data.
+
+    Parameters
+    ----------
+    config_data : dict
+        The configuration data
+
+    Returns
+    -------
+    rest_time : float
+        How long the starting voltage will be held in seconds.
+
+    """
+    
     rest_time = config_data["general_parameters"]["rest_time"]
 
     return rest_time
     
 def get_steps(config_data):
```

Comment:
I still don't quite get why we are having functions that just retrieve dictionary values.....   I guess it can make sense if it is part of our user-interface, and we don't expect our users to know python very well.  Is that the case?

Reply:
Though I expect users won't know python well, this part won't really be front-end. I envision in the end, user will enter into visual boxes which, when starting experiment, will write a config (which can be loaded if desired to conduct the same experiment later).  I just thought compartmentalizing the experimental config reading here might be good so these functions can read in and then go through a series of checks to make sure they are appropriate values (I still don't have these checks in for most of these functions) instead of bundling that with the functions of the other modules. I can just remake it and place the reading and the checks in the other modules where they are used though if that makes more sense.

Reply:
I see.  Actually, this is part of the MVC pattern and it would be the controller layer that does the validation, but it would probably be structured differently with a few validation functions that are more abstract (think about validating file paths, emails or passwords), but here we would be validating that floats are within some range or sthg and then return true or false and it would have statements like
```
if validate_voltage(config_data["ca"]["initial_voltage"], -5., 5.):
   ca_initial voltage = config_data["ca"]["initial_voltage"]
```

We can certainly do it your way, but we are writing a lot of code and a way that does it robustly with less code is probably preferred. 

Maybe we can discuss this a bit more on the call and make a decision. it will also depend on the UC a bit, so spending time hardening this code (docstrings and tests) may not be the best just yet (though I am glad this is now your impulse and wanting the code to validate!).  I think it has also been helpful for learning.  I am very excited about how you are now thinking about the coding!

Reply:
Okay yeah this makes sense, let's discuss this in the next meeting. Especially because we were going to discuss MVC anyways!